test.disabled/ejabberd_tests/tests/mam_SUITE.erl

```diff
@@ -1347,6 +1351,38 @@ long_text_search_request(Config) ->
         end,
     escalus_fresh:story(Config, [{alice, 1}, {bob, 1}], F).
 
+save_unicode_messages(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice, Bob) ->
+                escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"Hi! this is an unicode character ȥ"/utf8>>)),
+                escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"this is another one ȸ"/utf8>>)),
+                escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"This is the same again ȥ"/utf8>>)),
+                maybe_wait_for_archive(Config),
+
+                %% 'ȥ' query
+                escalus:send(Alice, stanza_text_search_archive_request(P, <<"q1">>, <<"ȥ"/utf8>>)),
+                Res1 = wait_archive_respond(P, Alice),
+                assert_respond_size(2, Res1),
+                assert_respond_query_id(P, <<"q1">>, parse_result_iq(P, Res1)),
+                [Msg1, Msg2] = respond_messages(Res1),
+                #forwarded_message{message_body = Body1} = parse_forwarded_message(Msg1),
+                #forwarded_message{message_body = Body2} = parse_forwarded_message(Msg2),
+                ?assert_equal(<<"Hi! this is an unicode character ȥ"/utf8>>, Body1),
+                ?assert_equal(<<"This is the same again ȥ"/utf8>>, Body2),
+
+                %% 'ȸ' query
+                escalus:send(Alice, stanza_text_search_archive_request(P, <<"q2">>, <<"ȸ"/utf8>>)),
+                Res2 = wait_archive_respond(P, Alice),
+                assert_respond_size(1, Res2),
+                assert_respond_query_id(P, <<"q2">>, parse_result_iq(P, Res2)),
+                [Msg3] = respond_messages(Res2),
+                #forwarded_message{message_body = Body3} = parse_forwarded_message(Msg3),
+                ?assert_equal(<<"this is another one ȸ"/utf8>>, Body3),
+
```

Comment:
Don't forget to test stuff that requires surrogates when encoded in UTF16.
For example:
𐀀 (Unicode Linear B Syllable B008 A)

```erlang
unicode:characters_to_binary([65536]).
<<240,144,128,128>>
```

And smiles 
https://en.wikipedia.org/wiki/Emoticons_(Unicode_block)

Reply:
The tested characters are already multibyte, and the fix/test are about unicode characters storage not about (currently not-our-implementation-defined) search capabilities.

Reply:
At least in mysql emoticons were broken without mb4 encoding (while lower multibyte code points were working just fine), so it's a real case.
It we have a test with them, than we would be sure that it works with all of our backends.

Search is completely different topic...